docs/spec/v1/buckets.md

```diff
@@ -651,6 +651,42 @@ When a Bucket's `.spec.provider` is set to `gcp`, the source-controller will
 attempt to communicate with the specified [Endpoint](#endpoint) using the
 [Google Client SDK](https://github.com/googleapis/google-api-go-client).
 
+The `gcp` provider can be used to authenticate automatically using OAuth scopes
+or Workload Identity, and by extension gain access to Google Cloud Storage.
```

Comment:
```suggestion
The `gcp` provider can be used to authenticate automatically using Workload Identity,
and by extension gain access to Google Cloud Storage.
```

I'd not mention `OAuth scopes` anywhere in these docs, they're kind of a very negligible implementation detail, we don't even use specific scopes per service in the implementation, we actually use single a wildcard scope for all of them:

`https://www.googleapis.com/auth/cloud-platform`